File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/klaytnDebug/runtimeDebugging/DebugMetricsTest.java
package opensdk.sdk.apis.klaytnDebug.runtimeDebugging;

import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.DebugMetricsResponse;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;

import java.io.IOException;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNull;

@DisplayName("Debug RPC Test")
public class DebugMetricsTest {
    private final OpenSDK sdk = new OpenSDK(UrlConstants.SERVER_URL);

    @Test
    @DisplayName("RPC debug_metrics")
    void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
        boolean raw = true;

        DebugMetricsResponse response = sdk.debug.metrics(raw).send();

        assertNotNull(response);
        assertNull(response.getError());
    }
}

File: web3rpc/sdk/client/javascript/openapi-test/test/debug/runtimeDebugging/DebugMetrics.test.js
const OpenSdk = require("opensdk-javascript");
const { expect } = require("@jest/globals");
const { RPC } = require("../../constant");

const sdk = new OpenSdk(new OpenSdk.ApiClient(RPC));

describe('debug_metrics API', () => {
    test('should return debug_metrics', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data).toBeDefined();
            expect(typeof data === 'object').toBe(true);
            done();
        };

        const raw = true;

        sdk.debug.metrics(raw, {}, callbackOne);
    });
});

